### PR TITLE
API tokens manager

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Created a manager and utilities around it for abstracting the handling of API tokens

### CODE REVIEW

File: src/db/models/api_token.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+use diesel::Queryable;
+
+#[allow(dead_code)]
+#[derive(Queryable)]
+#[diesel(table = "api_tokens")]
+pub struct APIToken {
+    token: String,
+    creation_time: String,
+    expire_time: String,
```
Comment (NicolasDP) [inline src/db/models/api_token.rs:9]: why are we using `String`? Why not `[u8; 32]`, `u64` and `u64` ?

File: src/v0/api_token.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+use diesel::query_dsl::{filter_dsl::FilterDsl, RunQueryDsl};
+use diesel::{ExpressionMethods, OptionalExtension};
+use warp::{Filter, Rejection};
+
+/// Header where token should be present in requests
+const API_TOKEN_HEADER: &str = "API-Token";
+
+/// API Token wrapper type
+#[derive(PartialEq, Eq)]
+pub struct APIToken(String);
```
Comment (NicolasDP) [inline src/v0/api_token.rs:16]: For the API Token, us a `[u8; 32]`. We will figure out the crypto for this later. In the header, indeed, encode it in base64 URL

File: src/db/models/api_token.rs (hunk 1 above)
Comment (danielSanchezQ) [reply src/db/models/api_token.rs:9]: Yes, the token should be some kind of hash. For the dates, it is actually to keep coherence with the other types we are currently using (encoded as strings).

Comment (danielSanchezQ) [reply src/db/models/api_token.rs:9]: Also for the current backend having dates as strings makes easy to make some queries:
https://sqlite.org/lang_datefunc.html

**Comment by danielSanchezQ** [comment]
I'm merging this, it works. More changes are coming in the next iterations and they will make an impact here too.

**Pull request merged**